src/theory/arith/theory_arith_private.cpp

```diff
@@ -3641,31 +3586,23 @@ void TheoryArithPrivate::postCheck(Theory::Effort effortLevel)
     }
   }//if !emmittedConflictOrSplit && fullEffort(effortLevel) && !hasIntegerModel()
 
-  if(!emmittedConflictOrSplit && effortLevel>=Theory::EFFORT_FULL){
-    if (d_nonlinearExtension != nullptr)
-    {
-      d_nonlinearExtension->check( effortLevel );
-    }
-  }
-
-  if(Theory::fullEffort(effortLevel) && d_nlIncomplete){
-    setIncomplete();
-  }
-
   if(Theory::fullEffort(effortLevel)){
     if(Debug.isOn("arith::consistency::final")){
       entireStateIsConsistent("arith::consistency::final");
     }
-    // cout << "fulleffort" << getSatContext()->getLevel() << endl;
-    // entireStateIsConsistent("arith::consistency::final");
-    // cout << "emmittedConflictOrSplit" << emmittedConflictOrSplit << endl;
   }
 
   if(Debug.isOn("paranoid:check_tableau")){ d_linEq.debugCheckTableau(); }
   if(Debug.isOn("arith::print_model")) {
     debugPrintModel(Debug("arith::print_model"));
   }
   Debug("arith") << "TheoryArithPrivate::check end" << std::endl;
+  return emmittedConflictOrSplit;
+}
+
+bool TheoryArithPrivate::isNonlinearIncomplete() const
+{
+  return d_nlIncomplete;
```

Comment:
Why does this piece of information still live in `TheoryArithPrivate`?

Reply:
This flag indicates whether `TheoryArithPrivate` is incomplete when used in isolation.  In other words, this tracks if the linear solver encountered any non-linear facts.

This is required when the non-linear extension is disabled.  This would hypothetically catch cases where non-linear arithmetic was accidentally introduced in a linear logic or in any other case where the non-linear extension is disabled.

Reply:
Ah, I see. The current name seems to suggest that the (computation of the) nonlinear extension was incomplete. Can we rename the flag? (like `foundNonlinearity`?)